engine/lib/notification.php

```diff
@@ -619,34 +618,67 @@ function elgg_send_email($from, $to, $subject, $body, array $params = null) {
 	}
 
 	// strip name from to and from
-
 	$to_address = Address::fromString($result['to']);
 	$from_address = Address::fromString($result['from']);
 
-
+	// make subject
 	$subject = elgg_strip_tags($result['subject']);
 	$subject = html_entity_decode($subject, ENT_QUOTES, 'UTF-8');
 	// Sanitise subject by stripping line endings
 	$subject = preg_replace("/(\r\n|\r|\n)/", " ", $subject);
 	$subject = trim($subject);
-
-	$body = elgg_strip_tags($result['body']);
-	$body = html_entity_decode($body, ENT_QUOTES, 'UTF-8');
-	$body = wordwrap($body);
-
-	$message = new Message();
+	
+	// build email message
+	$message = new EmailMessage();
 	$message->setEncoding('UTF-8');
 	$message->addFrom($from_address);
 	$message->addTo($to_address);
 	$message->setSubject($subject);
-	$message->setBody($body);
+	
+	// make the email body
+	$mime_body = new MimeMessage();
+	
+	// make plaintext message part
+	$plaintext = elgg_view('core/mail/plaintext', [
+		'subject' => $subject,
+		'body' => $result['body'],
+	]);
+	if (!empty($plaintext)) {
+		$plain_part = new MimePart($plaintext);
+		$plain_part->setType(Mime::TYPE_TEXT);
+		$plain_part->setCharset('UTF-8');
+		
+		$mime_body->addPart($plain_part);
+	}
+	
+	// make html message part
+	$html = elgg_view('core/mail/html', [
+		'subject' => $subject,
+		'body' => $result['body'],
+	]);
+	if (!empty($html)) {
+		// CSS inliner for email clients (eg Outlook, Thunderbird)
+		$html = _elgg_notifications_css_inliner($html);
+		
+		$html_part = new MimePart($html);
+		$html_part->setType(Mime::TYPE_HTML);
+		$html_part->setCharset('UTF-8');
+		
+		$mime_body->addPart($html_part);
+	}
+	
+	// set message body
+	$message->setBody($mime_body);
 
 	foreach ($result['headers'] as $headerName => $headerValue) {
 		$message->getHeaders()->addHeaderLine($headerName, $headerValue);
 	}
-
+	
+	// set content type correctly
+	$message->getHeaders()->get('content-type')->setType('multipart/alternative');
```

Comment:
You can leave this out. Zend will set content type based on the email parts. Setting the type explicitly was causing problems for me with Hotmail, because I was using only html part with attachments.